scripts/create_github_releases/main.go

```diff
@@ -0,0 +1,125 @@
+	// yes or no
+	validYesNo = regexp.MustCompile(`^(yes|no)$`)
+)
+
+func scanInputFromStdio(policy *regexp.Regexp, format *string, question, errMsg string) error {
+	fmt.Println(question)
+	fmt.Print("> ")
+	fmt.Scanf("%s", format)
+
+	// validate policy
+	if !policy.Match([]byte(*format)) {
+		return errors.New(errMsg)
+	}
+	return nil
+}
+
+// validateRelease validates policies for releases
+// used for release candidates. Return target branch name
+// based on the specified tag name.
+func validateRelease(tag, preRelease string) (string, error) {
+	if strings.Contains(tag, "-rc") {
+		// release candidates must be released as prereleases
+		if preRelease == "false" {
+			return "", errors.New("[ERROR] Prerelease must be true for release candidates")
+		}
+		// release candidates must use main branch as a target
+		return "main", nil
+	} else {
+		versions := strings.Split(tag, ".")
+		// naming convention for release branch is release-{major}-{minor}
+		return fmt.Sprintf("release-%s-%s", versions[0], versions[1]), nil
+	}
+}
+
+func main() {
+	fmt.Println("Start creating new GitHub Release interactively!")
+
+	var tag, target, preRelease, confirm string
+
+	if err := scanInputFromStdio(
+		validTag,
+		&tag,
+		"Q. Input version tag with the format {major}.{minor}.{patch} or {major}.{minor}.{patch}-{rc#} (e.g. 1.19.0 or 1.19.0-rc1)",
+		"[ERROR] Version tag must follow the format {major}.{minor}.{patch} or {major}.{minor}.{patch}-{rc#}",
+	); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	if err := scanInputFromStdio(
+		validPrerelease,
+		&preRelease,
+		"Q. Is prerelease? (Set true for release candidates) (e.g. true or false)",
+		"[ERROR] Prerelease must be true or false",
+	); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	target, err := validateRelease(tag, preRelease)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	title := fmt.Sprintf("Release %s", tag)
+	// Encode white space with percent encoding
+	// because url.QueryEscape does not support
+	// percent encoding.
+	title = strings.Replace(title, " ", "%20", -1)
+
+	repositoryURL := "https://github.com/cookpad/terraform-aws-eks"
+
+	// Check inputs
+	fmt.Println("# Check input to be used for a GitHub Release")
+	fmt.Println("tag:", tag)
+	fmt.Println("target branch:", target)
+	fmt.Println("pre release:", preRelease)
+	fmt.Println("title:", title)
+	fmt.Println()
+
+	if err := scanInputFromStdio(
+		validYesNo,
+		&confirm,
+		"Are these inputs correct? (yes or no)",
+		"[ERROR] Invalid confirmation",
+	); err != nil {
+		log.Fatalln(err.Error())
+	}
+	if confirm != "yes" {
+		fmt.Println("Canceled interactive GitHub release creation.")
+		os.Exit(0)
+	}
+
+	// https://docs.github.com/en/repositories/releasing-projects-on-github/automation-for-release-forms-with-query-parameters
+	url, err := url.Parse(
+		fmt.Sprintf("%s/releases/new?tag=%s&target=%s&title=%s&prerelease=%s", repositoryURL, tag, target, title, preRelease),
+	)
```

Comment:
```
url, err := url.Parse( fmt.Sprintf("%s/releases/new", repositoryURL))
...
q := url.Query()
q.Add("tag", tag)
...
u.RawQuery = q.Encode()
```

(and then you don't need to worry about https://github.com/cookpad/terraform-aws-eks/pull/264/files#diff-b2e438117526510a85bf4b119d773c8c48941505d8dc1ccce5a244f59180b754R86-R89)

Reply:
@aidy thanks! Your suggestion produces the following outputs

```
 https://github.com/cookpad/terraform-aws-eks/releases/new?tag=1.19.2&target=release-1-19&title=Release+1.19.2&prerelease=false
```

in the URL white space becomes `+` which I don't expect as an URL encoding. I'd rather choose a percent encoding since this is not the URL generated by forms. If we are fine to use `+`, 

```
	url, err := url.Parse(
		fmt.Sprintf("%s/releases/new?tag=%s&target=%s&title=%s&prerelease=%s", repositoryURL, tag, target, url.QueryEscape(title), preRelease),
	)
```

is also alternative. Do you have ideas to achieve the percent encoding for white spaces?

Reply:
🤔 The `+` encoding works when I click the link so 🤷 I guess it is OK if it makes the code simple!

Reply:
sounds reasonable! ♻️  https://github.com/cookpad/terraform-aws-eks/pull/264/commits/0e69b952bf107bb725b8c2ad54a38f20aaed6ee8